src/EventQueue.hs

```diff
@@ -36,28 +38,34 @@ emitDone chan = atomically $ writeTChan chan Done
 
 readEvents :: EventQueue -> IO [Event]
 readEvents chan = do
-  x <- atomically $ readTChan chan
-  threadDelay 100000
-  xs <- atomically $ emptyQueue
-  return (x : xs)
+  e <- atomically $ readTChan chan
+  unless (isKeyboardInput e) $ do
```

Comment:
Is it that you prefer `when` over `unless`?  In that case I would prefer `when (not . isKeyboardInput $ e)`.

Personally, I don't see anything wrong with `unless` though + `hlint` would warn about it.